tfx/orchestration/experimental/kubernetes/yaml/roles.yaml

```diff
@@ -0,0 +1,13 @@
+apiVersion: rbac.authorization.k8s.io/v1
```

Comment:
I think it is possible to specify a service account in the pod manifest: https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/. We would need to 1) Add instructions for users to create a new service account with edit privileges, and 2) Allow users to pass this service account through Kubernetes Dag Runner. However, it would be trickier to use different service accounts for different components without refactoring Kubernetes Component Launcher, since the current logic uses the Kubeflow service account for the Kubeflow case and the default one for the non-Kubeflow case.

Reply:
The issue is that applying this role will permanently (until explicitly removed) expand the privileges of the default service account for all Kubernetes pods.  Since people run not-completely-trusted code all the time, this introduces a vulnerability into the cluster configuration which will persist even after someone may have stopped using TFX on their cluster.  We should therefore scope this access to the services we create here (for example, by incorporating the service account config into this file):

```
apiVersion: v1
kind: ServiceAccount
metadata:
 name: my-service-account
```

and having that be granted the extra permissions.  Instead of using the default service account then in the KubernetesComponentLauncher, can we put everything we create (either via YAML specs or programatically) in this namespace instead?

Reply:
Thanks for the suggestion, this issue is now addressed by creating a new service account called tfx-service-account, and having the KubernetesComponentLauncher and ephemeral orchestrator job launch pods with this service account attached.